onnxscript/ir/_external_data.py

```diff
@@ -51,3 +70,245 @@ def set_base_dir(graph: _core.Graph | _core.GraphView, base_dir: str | os.PathLi
+                    external_tensor.base_dir = external_tensor.base_dir + "/temp/"  # type: ignore[operator]
+                raw_data = external_tensor.tobytes()
+            else:
+                raw_data = value.const_value.tobytes()
+            if current_offset is not None:
+                # Pad file to required offset if needed
+                file_size = data_file.tell()
+                if current_offset > file_size:
+                    data_file.write(b"\0" * (current_offset - file_size))
+            data_file.write(raw_data)
+
+
+def _store_as_external_tensors(
+    model: _core.Model,
+    external_data_info: list[tuple[_core.Value, ExternalDataInfo]],
+    file_path: str | os.PathLike,
+) -> _core.Model:
+    """
+    Convert the tensors (stored within the values) written as external data to _core.ExternalTensor types.
+
+    Args:
+        model: Model to be converted.
+        external_data_info: A collection of external data information stored for each tensor to be written as external data.
+        file_path: Location to which external data is to be stored.
+    """
+    for value, tensor_info in external_data_info:
+        tensor = value.const_value
+        assert tensor is not None
+        external_tensor = _core.ExternalTensor(
+            file_path,
+            tensor_info.offset,
+            tensor_info.length,
+            tensor.dtype,  # type: ignore[arg-type]
+            shape=tensor.shape,  # type: ignore[arg-type]
+            name=tensor.name,  # type: ignore[arg-type]
+        )
+        value.const_value = external_tensor
+        assert value.name in model.graph.initializers
+        model.graph.initializers[value.name] = value
+    return model
+
+
+def to_external_data(
+    model: _core.Model,
+    base_path: str | os.PathLike,
+    file_path: str = "",
+    align_offset: bool = True,
+    align_threshold: int = 1048576,  # 1MB,
+    allocation_granularity: int = 65536,  # 64KB
+    load_external_to_memory: bool = False,
+) -> _core.Model:
+    """
+    Call to set all tensors with raw data as external data.
+
+    Args:
+        model: Model to be converted.
+        base_path: Path of base directory.
+        file_path: Path to which external data is to be stored.
+        align_offset: Offset will always be page aligned and alloction granularity aligned for mmap support. This is done by padding previous tensor data with zeros keeping same length. Tensor data will be aligned if > align_threshold
+        align_threshold: Alignment threshold for size of data. Having a low threshold will waste file space for small initializers. Only when tensor's data is > the page_align_threshold it will be force aligned.
+        allocation_granularity: The allocation Granularity for mmap() support. Typically 64KB for Windows & 4KB for other OSes.
+        load_external_to_memory: If set to true, loads all pre-existing external tensors to memory. Otherwise, all new external tensors are appended to file.
```

Comment:
```suggestion
        load_external_to_memory: If set to true, loads all pre-existing external tensors to memory. Otherwise, all new external tensors are appended to file.
```

we don't need to load "all" external tensors. Only the conflicting ones.

> Otherwise, all new external tensors are appended to file.

I would update to make accurate